yet-another-docker-plugin/src/main/java/com/github/kostyasha/yad/commons/DockerPullImage.java

```diff
@@ -199,17 +198,9 @@ protected boolean shouldPullImage(DockerClient client, String imageName) {
             return false;
         }
 
-        List<Image> images = client.listImagesCmd().exec();
+        List<Image> images = client.listImagesCmd().withImageNameFilter(fullImageName).exec();
 
-        boolean hasImage = images.stream().anyMatch(image ->
-                nonNull(image.getRepoTags()) &&
-                        Arrays.stream(image.getRepoTags())
-                                .anyMatch(repoTag ->
-                                        repoTag.contains(fullImageName) || repoTag.contains("docker.io/" + fullImageName)
-                                )
-        );
-
-        return hasImage ?
+        return images.isEmpty() ?
```

Comment:
hasImage = true, that means that image exists
isEpmty() = true mean that imagee doesn't exist